### ISSUE TITLE
Modular Digigrade Legs have incorrect Essence cost

### ISSUE THREAD

**Issue by ResonantGhost** [chummer5a/chummer5a#3525]
**Describe the bug**
A modular limb costs no essence instead of 1 essence (using full limb for example). Since modular limbs are alternatives to using a normal limb with a modular connector, the first one should cost essence.

**To Reproduce**
Prerequisite: editing a character. (For simplicity, I am editing a new character)
1. Click on Cyberware and Bioware tab.
2. Click Add Cyberware.
3. Select Obvious Full Arm, Modular
4. Click Ok
5. Click Ok (for default left arm)
6. See that the Essence is still 6.0

**Expected behavior**
Essence should be reduced by 1.0 for installing a modular limb.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 5.205.13

**Additional context**
Issue appears to stem from [feature request 900](https://github.com/chummer5a/chummer5a/issues/900). However, the modular connector is for use with a standard limb, not modular limbs. See post from Wakshaani clarifying modular limbs versus modular connectors: https://forums.shadowruntabletop.com/index.php?topic=26233.msg487011#msg487011

For the time being, I am adding an amend_cyberware.xml that adds an installed modular limb with the correct essence cost. If this above link is not sufficient to establish official behavior, I can open a feature request for Wakshaani's implementation, but the 0 essence cost of modular limbs is still incorrect.

**Comment by ResonantGhost** [comment]
Attached is XML file added to my customdata to resolve issue (I hope).

[amend_cyberware.zip](https://github.com/chummer5a/chummer5a/files/3001493/amend_cyberware.zip)

**Comment by chummer5a** [comment]
That won't actually do what you want it to; the way we implemented modular limbs, you install a modular connector that has a total cost of the limb that is removed plus the modular connector for that location. An arm costs 1 essence plus .3 for the connector. The modular arm itself then costs 0 essence, because the connector's the one doing all the work. 

A Modular limb comes with its side of the modular connector pre-installed for a reduction in capacity. A regular limb would need to have its side of the modular connector installed for a separate cost. Both still need to be connected to a modular connector that's in your meat.

**Comment by ResonantGhost** [comment]
That's why I linked Wakshaani's post. The modular limb does not need a modular connection implant. 

It's built into the arm instead of your shoulder (and so the modular limb has less capacity). If you use the modular connector, you use a standard limb instead with full capacity, but you pay more essence.

I may be putting too much importance on his ability to speak on the rules, but I believe feature request 900 was implemented incorrectly.

**Comment by cfresquet** [comment]
Modular Limb (p88 of Chrome Flesh)
--------------------------
Any cyberlimb may be purchased with a modular connector, allowing it to be CONNECTED TO ANOTHER MODULAR CONNECTOR (emphasis mine).

You need the modular connector on both sides of the connection.  It is implemented correctly.

**Comment by ResonantGhost** [comment]
See the post by Wakshaani above. Wakshaani is Thomas Willoughby, Freelance author  on Chrome Flesh.
This is also consistent with the Chrome Flesh Errata post, again by Wakshaani:
https://forums.shadowruntabletop.com/index.php?topic=21221.msg383327#msg383327

It is also consistent with the cyberware pattern:
Hydraulic Jacks, Telescopic limbs, Water Jets, etc. can only be installed in a cyber limb and have a capacity cost with no essence cost. If modular limbs could only be installed on a modular connector, which covers the essence cost of that limb, the modular limb would have a zero essence cost. However, modular limbs have an essence cost which implies that it contains both sides of the connector. Connecting to another connector does not rule out possessing it's own connector and connecting to that.

The implemented interpretation ignores the documented essence cost and is not consistent with the pattern of essence costs of cyberware. As long as there is no scenario where a modular limb costs essence, it is not an implementation of a literal interpretation of the rules. It may be an agreeable work around for requested feature.

There is some dispute whether the posts, clarifications or otherwise by Catalyst's team is sufficient to constitute "errata." Outside of the core rulebook, I have not found any compiled errata documents, only the errata threads, where Waskshaani posted answers and clarifications for Chrome Flesh.

If this is not sufficient for the Chummer team's purposes, I have the above XML file which appears to be working as expected.

**Comment by cfresquet** [comment]
> modular limbs have an essence cost which implies that it contains both sides of the connector

No, it doesn't.  (A) The text I quoted explicitly says that it doesn't.  The word "a" is singular. (B) The essence cost is exactly the same for both modular and non-modular.

The fact that the author has posted in an unofficial errata thread saying that what was published doesn't match his intended design doesn't change what the book says.  

Until CGL puts out an official or at least semi-official (and publicly posted somewhere) collected set of errata for Chrome Flesh I don't think Chummer should be basing its rules interpretations off of unsanctioned 4-year-old forum posts that the vast majority of players and GM's have no access to or desire to read.

That being said, I have zero problem with an optional or house rule implementing this: But without official errata I would strongly suggest sticking to RAW whenever possible.

**Comment by chummer5a** [comment]
The complaint I have with how wakshaani thinks his rules work is that if that's the case, there is no scenario in which one would actually want the separate modular connector cyberware, aside from it gating a limb beyond Avail 12 with decentish limb attributes.

So. I won't have time to properly test your customdata for a while, but at first blush I would expect to see that you will not be able to detach a limb from your body and you MIGHT still be able to purchase additional arms, but they will definitely continue to deduct Essence.

The least-ghetto fix I could propose for you would be to set the Essence cost of Modular Connectors to the cost of the limb, ie 1 for a modular limb. There aren't any conditionals for whether a mount can take a given mount type so you'd have to give them some kind of name to indicate they're for modular limbs only.

**Comment by ResonantGhost** [comment]
@cfresquet, I do not feel this is the appropriate place to debate modular limbs. I am satisfied with Wakshaani's clarification and do not see an errata as necessary. If you decide you want to discuss it on dumpshock, the official forums, or reddit, I would be happy to join the discussion there.

@chummer5a, I am not sure how you mean that it won't do what I expect. It appears to be working as expected, but you may see something I missed.

**Comment by cfresquet** [comment]
@ResonantGhost : you just summed up my issue with Wakshaani's post -- it isn't a clarification, it is a change.

I don't have a problem with clarifications.  Changes, on the other hand, should be discouraged unless they are official errata, because the vast majority of people won't ever see an unofficial forum post.

**Comment by ResonantGhost** [comment]
@cfresquet, as mentioned, I don't think we should debate this here. I have added [this post to the forums](https://forums.shadowruntabletop.com/index.php?topic=29067.0). Feel free to ensure that I have accurately represented your opinion.

**Comment by ResonantGhost** [comment]
Based on your comments in the forum post, I tried playing with the modular limbs (stock, not my modifications) in career mode and couldn't figure out how to add them to a character. I also wanted to see what happened if you swapped between digigrade and regular cyber legs (would essence fluctuate?), but I couldn't get it to work at all.

I couldn't add the modular limbs to the connector, nor could I change the mounting location. Clicking change mounting location pops up two windows that immediately disappear. (one, in the center that is probably meant to be a modal prompt of potential locations and a smaller on to the upper left) I never see the contents of the popup render.

Is change mounting location how I would attach modular limbs?

**Comment by chummer5a** [comment]
So what you should get is something like this:

![image](https://user-images.githubusercontent.com/11448441/55277570-e1f89100-5355-11e9-9b09-ebfacce3ade2.png)

Is that not the case?

**Comment by ResonantGhost** [comment]
No, it disappears immediately. It never appears to actually have content. When I have a chance, I will try it on my other laptop that has an earlier version, try reinstalling, etc. to see if I can make it show or let you know when it stops working. I just wanted to check if I was doing it right.

My set up was two hip connectors, left and right and four modular legs, two left and two right. One left and one right were digigrade. The others were basic.

**Comment by ResonantGhost** [comment]
So, I managed to get it working. Sort of. I don't know why it works apart from maybe the exact steps I took are different. The differences I know are different are: I added an extra set of synthetics; and I added limbs to the hip connectors before marking the character as created.

Quirks/Potential Bugs:
- Modular synthetic legs do not require that you specify left or right.
- Adding a modular limb to a hip joint causes the hip joint to exceed its capacity and you appear to be able to save in career mode, but not create mode.
- Having modular digigrade legs is cool, but switching between those and regular modular limbs causes your essence to fluctuate. 2 hip connectors is 2.6 Essence which should leave me with 3.4 essence, but when I attach 2 obvious limbs, it puts me at 2.9 Essence. If I swap one out for a digigrade leg, it goes up to 3.15. If I add a second digigrade leg, it will go up to 3.4. Given the increased cost of the digigrade legs, I would expect Essence to remain at 2.9.

For my personal house rules, I may drop the essence cost of the connectors or I may ban modular limbs. I don't really have anyone expressing an interest in them at this time. Separately, I suspect that if I flatten the essence cost for digigrade versus normal legs, the essence costs will stop fluctuating.

**Comment by chummer5a** [comment]
Fixed the issue with with limb capacity and synthetic limbs not prompting for sides. Modular digigrade is pretty much nonviable with our implementation, largely because the modular limbs themselves aren't permitted to add to their parent's essence. From memory, if we allow it to add to parent essence it counts the children's essence while the limb is disconnected. 

Can you show me a copy of your file with the essence fluctuations? That doesn't sound right at all.

**Comment by ResonantGhost** [comment]
I will try to upload it when I get home. The essence fluctuations were, I believe from the digigrade modification being added to the modular leg, which is not an official option, to my understanding. Based on what you have posted on your standards, changing digigrade to it's own cyber limb instead of a modification or forbidding it for modular limbs should prevent the issue.

Aside: if you have digigrade legs with a knee modular joint, can you wind up with normal legs that give the digigrade bonus? :)

**Comment by ResonantGhost** [comment]
It's a skeleton character, but see attached.

[Modular Tesing.zip](https://github.com/chummer5a/chummer5a/files/3042508/Modular.Tesing.zip)

**Comment by chummer5a** [comment]
That's about the sum of it, yeah. There's not really a good solution for it without code changes, but it's most likely fixable without too much hassle. I might not get a chance to look into it until wakshaani has his second crack at liminal bodies and such and we get some stealth errata out of it.
